lib/inference/hugging_face_text_generation.rb

```diff
@@ -45,6 +40,10 @@ def self.perform!(
         parameters[:max_new_tokens] = token_limit - prompt_size
         parameters[:temperature] = temperature if temperature
         parameters[:repetition_penalty] = repetition_penalty if repetition_penalty
+        
+        if block_given?
```

Comment:
Moving to headers instead of path makes it compatible with both self-hosting and hosted API